File: marktia-frontend/dtos/responses/users/userResponseDTO.ts
import { AddressResponseDTO } from "../address/AddressResponseDTO";

export type UserResponseDTO = {
    id: string;
    name: string;
    email: string;
    cpf: string;
    cnpj?: string;
    telephone: string;
    address: AddressResponseDTO;
    userRole: number;
    creationDate: string;
    updateDate?: string;
    imageURL?: string;
}

File: marktia-backend/src/main/java/ufes/marktiabackend/dtos/responses/UserResponseDTO.java
package ufes.marktiabackend.dtos.responses;

import lombok.*;
import ufes.marktiabackend.entities.Address;
import ufes.marktiabackend.enums.UserRole;

import java.time.LocalDate;

@Data
@Builder
public class UserResponseDTO {
    private String id;

    private String name;

    private String email;

    private String cpf;

    private String cnpj;

    private String telephone;

    private AddressResponseDTO address;

    private Integer userRole;

    private String creationDate;

    private String updateDate;

    private String imageURL;
}
